src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Confirm.tsx

```diff
@@ -175,6 +183,110 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
     state?.deposit.cryptoAmount,
   ])
 
+  const getPreDepositInput: () => Promise<SendInput | undefined> = useCallback(async () => {
+    if (!(accountId && assetId && state?.deposit?.estimatedGasCrypto)) return
+
+    try {
+      const estimatedFees = await estimateFees(await getEstimateFeesArgs())
+      const amountCryptoBaseUnit = bnOrZero(state?.deposit.cryptoAmount).times(
+        bn(10).pow(asset.precision),
+      )
+      const quote = await getThorchainSaversQuote(asset, amountCryptoBaseUnit)
+
+      const sendInput: SendInput = {
+        cryptoAmount: '',
+        asset,
+        from: '', // Let coinselect do its magic here
+        to: maybeFromUTXOAccountAddress,
+        sendMax: true,
+        accountId,
+        amountFieldError: '',
+        cryptoSymbol: asset.symbol,
+        estimatedFees,
+        feeType: FeeDataKey.Fast,
+        fiatAmount: '',
+        fiatSymbol: selectedCurrency,
+        vanityAddress: '',
+        input: quote.inbound_address,
+      }
+
+      return sendInput
+    } catch (e) {
+      moduleLogger.error({ fn: 'getDepositInput', e }, 'Error building THORChain savers Tx')
+    }
+  }, [
+    maybeFromUTXOAccountAddress,
+    accountId,
+    assetId,
+    getEstimateFeesArgs,
+    state?.deposit.cryptoAmount,
+    state?.deposit?.estimatedGasCrypto,
+    asset,
+    selectedCurrency,
+  ])
+
+  const handleMultiTxSend = useCallback(async (): Promise<string | undefined> => {
+    if (!walletState.wallet) return
+
+    // THORChain Txs need to always be sent from the same address, since the address (NOT the pubkey) is used to identify an active position
+    // The way THORChain does this is by not being xpub-compliant, and only exposing a single address for UTXOs in their UI
+    // All deposit/withdraws done from their UI are always done with one/many UTXOs from the same address, and change sent back to the same address
+    // We also do this EXCLUSIVELY for THORChain Txs. The rest of the app uses xpubs, so the initially deposited from address isn't guaranteed to be populated
+    // if users send other UTXO Txs in the meantime after depositing
+    // Additionally, we select their highest balance UTXO address as a first deposit, which isn't guaranteed to contain enough value
+    //
+    // For both re/deposit flows, we will possibly need a pre-Tx to populate their highest UTXO/previously deposited from address with enough value
+
+    const depositInput = await getDepositInput()
+    if (!depositInput) throw new Error('Error building send input')
+
+    let txId: string
+
+    try {
+      // 1. Try to deposit from the originally deposited from / highest UTXO balance address
+      // If this is enough, no other Tx is needed
+      txId = await handleSend({
+        sendInput: depositInput,
+        wallet: walletState.wallet,
+      })
+    } catch (e) {
+      // 2. signAndBroadcastTransaction threw, meaning there's not enough value in the picked address - send funds to it
```

Comment:
`signAndBroadcastTransaction` isn't strictly correct here - that's a term we use specifically for "online" wallets like metamask etc.

Reply:
a0129da8f14410ada108a75c65a6712dc0df157f